src/main/java/gov/cabinetoffice/gap/adminbackend/services/SchemeEditorService.java

```diff
@@ -81,5 +87,30 @@ private List<String> getEmailsFromUserSubBatch(final List<String> userSubs, fina
                 awsEncryptionService.decryptField(item.emailAddress())).toList();
     }
 
+    public SchemeEntity addEditorToScheme(Integer schemeId, String editorEmailAddress, String jwt) {
+        AdminSession session = HelperUtils.getAdminSessionForAuthenticatedUser();
+        SchemeEntity scheme = this.schemeRepo.findById(schemeId).orElseThrow(EntityNotFoundException::new);
+        List<GrantAdmin> existingEditors = scheme.getGrantAdmins();
+        GrantAdmin editorToAdd;
+        try {
+            editorToAdd = userService.getGrantAdminIdFromUserServiceEmail(editorEmailAddress, jwt);
+        } catch (Exception e) {
+            throw new FieldViolationException("editorEmailAddress", "This account does not have an 'Administrator' account.");
+        }
+
+        if (existingEditors.stream().anyMatch(editor -> editor.getId().equals(editorToAdd.getId()))) {
+            throw new FieldViolationException("editorEmailAddress", "This email address is already an editor for this scheme");
+        }
+
+        if (!scheme.getCreatedBy().equals(session.getGrantAdminId())) {
+            throw new AccessDeniedException(
+                    "You are unable to add an editor to the scheme with id " + schemeId);
+        }
+
+        scheme.addAdmin(editorToAdd);
+        this.schemeRepo.save(scheme);
+        return scheme;
```

Comment:
we should really be returning a fresh copy from the database so probably worth returning immediately on the line above